mavros_extras/src/plugins/obstacle_distance_3d.cpp

```diff
@@ -59,35 +59,36 @@
      }
  
  private:
-      rclcpp::Subscription<mavros_msgs::msg::Obstacle3D>::SharedPtr obstacle_sub_;
+      rclcpp::Subscription<mavros_msgs::msg::ObstacleDistance3D>::SharedPtr obstacle_sub_;
  
      mavlink::common::MAV_FRAME frame;
  
      /**
       * @brief Callback for incoming obstacle data. Converts and sends to FCU.
       *
-      * Message specification: https://mavlink.io/en/messages/common.html#OBSTACLE_DISTANCE_3D
-      * @param msg  The received Obstacle3D message.
-      */
-      void obstacle_cb(const mavros_msgs::msg::Obstacle3D::SharedPtr msg)
+      * Message specification: https://mavlink.io/en/messages/ardupilotmega.html#OBSTACLE_DISTANCE_3D
+     * @param msg  The received ObstacleDistance3D message.
+     */
+      void obstacle_cb(const mavros_msgs::msg::ObstacleDistance3D::SharedPtr msg)
      {
-         // --- Data Validation ---
-         // Check for invalid floating point numbers before sending.
-         if (std::isnan(msg->x) || std::isnan(msg->y) || std::isnan(msg->z)) {
-             RCLCPP_WARN(get_logger(), "obstacle_3d: Rejecting message with NaN coordinates.");
-             return;
-         }
+        // --- Data Validation ---
+        // Check for invalid floating point numbers before sending.
+        if (std::isnan(msg->position.x) || std::isnan(msg->position.y) || std::isnan(msg->position.z)) {
+            RCLCPP_WARN(get_logger(), "obstacle_3d: Rejecting message with NaN coordinates.");
+            return;
+        }
  
-         mavlink::ardupilotmega::msg::OBSTACLE_DISTANCE_3D obstacle{};
- 
-         // --- Field Mapping ---
-         obstacle.sensor_type = msg->sensor_type;
-         obstacle.obstacle_id = msg->obstacle_id;
-         obstacle.x = msg->x;
-         obstacle.y = msg->y;
-         obstacle.z = msg->z;
-         obstacle.min_distance = msg->min_distance;
-         obstacle.max_distance = msg->max_distance;
+        mavlink::ardupilotmega::msg::OBSTACLE_DISTANCE_3D obstacle{};
+
+        // --- Field Mapping ---
+        obstacle.time_boot_ms = get_time_boot_ms();
```

Comment:
```suggestion
        obstacle.time_boot_ms = get_time_boot_ms(msg->header.stamp);
```

https://github.com/search?q=repo%3Amavlink/mavros%20get_time_boot_ms&type=code